sycl/include/sycl/queue.hpp

```diff
@@ -122,10 +123,72 @@ class __SYCL_EXPORT queue {
   queue(const async_handler &AsyncHandler, const property_list &PropList = {})
       : queue(default_selector(), AsyncHandler, PropList) {}
 
+#if __cplusplus >= 201703L
+  /// Constructs a SYCL queue instance using the device identified by the
+  /// device selector provided.
+  /// \param DeviceSelector is SYCL 2020 Device Selector, a simple callable that
+  /// takes a device and returns an int
+  /// \param AsyncHandler is a SYCL asynchronous exception handler.
+  /// \param PropList is a list of properties for queue construction.
+  template <typename DeviceSelector,
+            typename = detail::EnableIfDeviceSelectorInvocable<
+                DeviceSelector, ext::oneapi::filter_selector>>
+  explicit queue(const DeviceSelector &deviceSelector,
+                 const async_handler &AsyncHandler,
+                 const property_list &PropList = {})
+      : queue(detail::select_device(deviceSelector), AsyncHandler, PropList) {}
+
+  /// Constructs a SYCL queue instance using the device identified by the
+  /// device selector provided.
+  /// \param DeviceSelector is SYCL 2020 Device Selector, a simple callable that
+  /// takes a device and returns an int
+  /// \param PropList is a list of properties for queue construction.
+  template <typename DeviceSelector,
+            typename = detail::EnableIfDeviceSelectorInvocable<
+                DeviceSelector, ext::oneapi::filter_selector>>
+  explicit queue(const DeviceSelector &deviceSelector,
+                 const property_list &PropList = {})
+      : queue(detail::select_device(deviceSelector), async_handler{},
+              PropList) {}
+
+  /// Constructs a SYCL queue instance using the device identified by the
+  /// device selector provided.
+  /// \param SyclContext is an instance of SYCL context.
+  /// \param DeviceSelector is SYCL 2020 Device Selector, a simple callable that
+  /// takes a device and returns an int
+  /// \param PropList is a list of properties for queue construction.
+  template <typename DeviceSelector,
+            typename = detail::EnableIfDeviceSelectorInvocable<
+                DeviceSelector, ext::oneapi::filter_selector>>
```

Comment:
I have the feeling you are calling it always with `ext::oneapi::filter_selector`, so why not moving it into the concept?
```suggestion
            typename = detail::EnableIfDeviceSelectorInvocable<DeviceSelector>
```

Reply:
Is needless overengineering a valid reason?  

Fixed.

Reply:
SYCL is all about elegance, style and beauty compared to non standard alternatives from the competition. ;-) Thanks.